syntax = "proto3";
import "google/protobuf/timestamp.proto";
package pb;

option go_package = "github.com/dxnnymurphy/gRPC/pb";

message Metric {
    google.protobuf.Timestamp time = 1;
    string topic = 2;
    string refreshbool = 3;
    string stop = 4;
}

message Metric1 {
    google.protobuf.Timestamp startTime = 1;
    google.protobuf.Timestamp endTime = 2;
}

message AnomalyRequest {
    repeated Metric metrics = 1;
}

message AnomalyResponse {
    repeated string response = 1;
}

message ModelTrainRequest {
    repeated Metric metrics = 1;
}

message ModelTrainResponse {
    repeated string response =1;
}

service AnomalyDetection {
    rpc Predict(AnomalyRequest) returns (AnomalyResponse) {}
    rpc Train(ModelTrainRequest) returns (ModelTrainResponse)
}